pkg/expansion/system.go

```diff
@@ -117,6 +142,40 @@ func (s *System) templatesForGVK(gvk schema.GroupVersionKind) []*expansionunvers
 // mutators. If no ExpansionTemplates match `base`, an empty slice
 // will be returned. If `s.mutationSystem` is nil, no mutations will be applied.
 func (s *System) Expand(base *mutationtypes.Mutable) ([]*Resultant, error) {
+	var res []*Resultant
+	if err := s.expandRecursive(base, &res, 0); err != nil {
+		return nil, err
+	}
+	return res, nil
+}
+
+func (s *System) expandRecursive(base *mutationtypes.Mutable, resultants *[]*Resultant, depth int) error {
```

Comment:
A pointer to a list of pointers? It's evil, I love it.

(Also necessary unless you want to return the list as a return value and feed it back into the function)

Reply:
I want to modify the slice inside the function. `append` returns a new list, so if I just pass the slice value it will only update the slice within the scope of the function. Demo: https://go.dev/play/p/CVwqHU5-ktP

Or are you suggesting I return the slice instead?

Reply:
No, this is right, just evil